src/commands/aksRetinaCapture/aksRetinaCapture.ts

```diff
@@ -0,0 +1,133 @@
+import { getKubernetesClusterInfo } from "../utils/clusters";
+import { getExtension, longRunning } from "../utils/host";
+import * as tmpfile from "../utils/tempfile";
+import path from "path";
+import { ensureDirectoryInPath } from "../utils/env";
+import { getRetinaBinaryPath } from "../utils/helper/retinaBinaryDownload";
+import { getVersion, invokeKubectlCommand } from "../utils/kubectl";
+import { RetinaCapturePanel, RetinaCaptureProvider } from "../../panels/RetinaCapturePanel";
+import { failed } from "../utils/errorable";
+
+export async function aksRetinaCapture(_context: IActionContext, target: unknown): Promise<void> {
+    const kubectl = await k8s.extension.kubectl.v1;
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    const clusterExplorer = await k8s.extension.clusterExplorer.v1;
+
+    if (!kubectl.available) {
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+        return;
+    }
+
+    if (!cloudExplorer.available) {
+        vscode.window.showWarningMessage(`Cloud explorer is unavailable.`);
+        return;
+    }
+
+    if (!clusterExplorer.available) {
+        vscode.window.showWarningMessage(`Cluster explorer is unavailable.`);
+        return;
+    }
+
+    const clusterInfo = await getKubernetesClusterInfo(target, cloudExplorer, clusterExplorer);
+    if (failed(clusterInfo)) {
+        vscode.window.showErrorMessage(clusterInfo.error);
+        return;
+    }
+
+    const kubectlRetinaPath = await getRetinaBinaryPath();
+    if (failed(kubectlRetinaPath)) {
+        vscode.window.showWarningMessage(`kubectl retina path was not found ${kubectlRetinaPath.error}`);
+        return;
+    }
+
+    ensureDirectoryInPath(path.dirname(kubectlRetinaPath.result));
+
+    const extension = getExtension();
+    if (failed(extension)) {
+        vscode.window.showErrorMessage(extension.error);
+        return;
+    }
+
+    const kubeConfigFile = await tmpfile.createTempFile(clusterInfo.result.kubeconfigYaml, "yaml");
+    console.log(`Kubeconfig file: ${kubeConfigFile}`);
+
+    // Get all Nodes For this Cluster
+    const nodelistResult = await invokeKubectlCommand(
+        kubectl,
+        kubeConfigFile.filePath,
+        `get nodes -o jsonpath="{range .items[*]}{.metadata.name}{' '}{end}"`,
+    );
+
+    if (failed(nodelistResult)) {
+        vscode.window.showErrorMessage(`Failed to capture the cluster: ${nodelistResult.error}`);
+        return;
+    }
+    const nodesList = nodelistResult.result.stdout.trim().split(" ").map((nodeName) => ({ label: nodeName }));
+
+    // Pick a cluster from group to compare with
+    const nodeNamesSelected = vscode.window.showQuickPick(nodesList, {
+        canPickMany: true,
+        placeHolder: "Please select all the Nodes you want Retina to capture traffic from.",
+        title: "Select Nodes to Capture Traffic From",
+    });
+
+    const selectedNodes = await nodeNamesSelected.then((selectedItems) => {
```

Comment:
To avoid dealing with `undefined` and an extra conversions between strings and arrays, you could do:
```typescript
const nodeNamesSelected = await ...;
if (!nodeNamesSelected) {
    return;
}

const selectedNodes = nodeNamesSelected.map(item => item.label);
```